java/client/src/main/java/glide/api/models/commands/bitmap/BitFieldOptions.java

```diff
@@ -0,0 +1,187 @@
+        public enum BitOverflowControl {
+            WRAP,
+            SAT,
+            FAIL
+        }
+    }
+
+    private interface BitEncoding {
+        String getEncoding();
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class SignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "i" to specify that it is in signed encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public SignedEncoding(long encodingLength) {
+            encoding = "i".concat(Long.toString(encodingLength));
+        }
+    }
+
+    /** Specifies that the argument is a signed encoding */
+    public static final class UnsignedEncoding implements BitEncoding {
+        @Getter private final String encoding;
+
+        /**
+         * Constructor that prepends the number with "u" to specify that it is in unsigned encoding.
+         *
+         * @param encodingLength bit size of encoding.
+         */
+        public UnsignedEncoding(long encodingLength) {
+            encoding = "u".concat(Long.toString(encodingLength));
+        }
+    }
+
+    private interface BitOffset {
+        String getOffset();
+    }
+
+    /** Offset in the array of bits. */
+    public static final class Offset implements BitOffset {
+        @Getter private final String offset;
+
+        /**
+         * Constructor for Offset.
+         *
+         * @param offset element in the array of bits.
+         */
+        public Offset(long offset) {
+            this.offset = Long.toString(offset);
+        }
+    }
+
+    /** Offset in the array of bits multiplied by the encoding value. */
+    public static final class OffsetMultiplier implements BitOffset {
+        @Getter private final String offset;
+
+        /**
+         * Constructor that prepends the offset value with "#" to specify that it is a multiplier.
+         *
+         * @param offset element multiplied by the encoding value in the array of bits.
+         */
+        public OffsetMultiplier(long offset) {
+            this.offset = "#".concat(Long.toString(offset));
+        }
+    }
+
+    /**
+     * Creates the arguments to be used in {@link BitmapBaseCommands#bitfield(String,
+     * BitFieldSubCommands[])} and {@link BitmapBaseCommands#bitfieldReadOnly(String,
+     * BitFieldReadOnlySubCommands[])}.
+     *
+     * @param subCommands commands that holds arguments to be included in the argument String array.
+     * @return a String array that holds the sub commands and their arguments.
+     */
+    public static String[] createBitFieldArgs(BitFieldSubCommands[] subCommands) {
+        String[] arguments = {};
+
+        for (int i = 0; i < subCommands.length; i++) {
+            arguments = concatenateArrays(arguments, subCommands[i].createArgs());
+        }
+
+        return arguments;
+    }
```

Comment:
Probably, you need a UT to prove that everything converted properly.

Reply:
Would love to be where do we add unit tests for non command functions? Was using `createZRangeBaseArgs` as reference and it's not being tested.

Reply:
Somewhere in UT (`:client:test`). Unfortunately, we're missing some tests. It is not a mandatory requirement, but good for coveraga and catching typos.